templating/templating.py

```diff
@@ -52,14 +55,18 @@ def add_header(response):
 
 @app.route("/<path>", methods=["GET"])
 def root_get(path):
-    updateData()
     try:
+        updateData()
         template = env.get_template(path)
         body = template.render(objects=objects, options=request.args)
     except TemplateNotFound:
         return 'Template "{}" not found\n'.format(path), 404
-    except Exception as err:
+    except TemplateError as err:
         return 'Templating of "{}" failed to render. Most likely due to an error in the template. Error transcript:\n\n{}\n----\n\n{}\n'.format(path, err, traceback.format_exc()), 400
+    except requests.exceptions.HTTPError as err:
+        return 'HTTP error from gondul: {}'.format(err), 500
+    except Exception as err:
+        return 'Connection issues: {}'.format(err), 500
```

Comment:
```suggestion
        return f'Uncaught error: {error}', 500
```
Could be misleading. A request where `.json()` fails due to a HTML body for instance would cast this error.